supportedBackends/tes/src/main/scala/cromwell/backend/impl/tes/TesAsyncBackendJobExecutionActor.scala

```diff
@@ -0,0 +1,261 @@
+package cromwell.backend.impl.tes
+
+import java.nio.file.{FileAlreadyExistsException, Path}
+
+import better.files.File
+import cromwell.backend.BackendJobLifecycleActor
+import cromwell.backend.async.{ExecutionHandle, FailedNonRetryableExecutionHandle, PendingExecutionHandle, SuccessfulExecutionHandle}
+import cromwell.backend.impl.tes.TesResponseJsonFormatter._
+import cromwell.backend.io.JobPaths
+import cromwell.backend.sfs.SharedFileSystemExpressionFunctions
+import cromwell.backend.standard.{StandardAsyncExecutionActor, StandardAsyncExecutionActorParams, StandardAsyncJob, StandardInitializationData}
+import cromwell.backend.wdl.OutputEvaluator
+import cromwell.core.path.FileImplicits._
+import cromwell.core.path.PathFactory.pathPlusSuffix
+import cromwell.core.path.{PathBuilder, PathFactory}
+import cromwell.core.retry.SimpleExponentialBackoff
+import lenthall.util.TryUtil
+import spray.client.pipelining._
```

Comment:
To throw another wrench in the mix, another idea is to use a library generated from the protobuf/grpc task execution schemas. Seems like possibly everyone is missing out on one of the primary benefits of using protobuf and grpc by implementing everything as http/json.